### ISSUE TITLE
Support for Expressroute Direct circuits

### ISSUE THREAD

**Issue by bardahlm** [terraform-providers/terraform-provider-azurerm#11155]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Setting up Expressroute Direct circuits require some other parameters than circuits using a service provider:

* bandwidthInGbps, not bandwidthInMbps
* expressRoutePort
* portType should be false
and possible some other parameters that need to be adjusted.

### New or Affected Resource(s)

* azurerm_express_route_circuit

### Potential Terraform Configuration

```hcl
resource "azurerm_express_route_circuit" "example" {
  name                  = "expressRoute1"
  resource_group_name   = azurerm_resource_group.example.name
  location              = azurerm_resource_group.example.location
  service_provider_name = "Microsoft.Network"
  peering_location      = "Silicon Valley"
  bandwidth_in_Gbps     = 1
  express_route_port_id = "/subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/exprtTest/providers/Microsoft.Network/expressRoutePorts/er-exampleport"

  sku {
    tier   = "Standard"
    family = "MeteredData"
  }

  tags = {
    environment = "Production"
  }
}
```

### References

* https://docs.microsoft.com/en-us/rest/api/expressroute/expressroutecircuits/createorupdate
* https://docs.microsoft.com/en-us/azure/expressroute/quickstart-create-expressroute-vnet-template

**Comment by ArielRam99** [comment]
@tombuildsstuff Is there work being done on this issue in the background?

**Comment by LaurentLesle** [comment]
Is there a way to fast track this resource?

**Comment by LaurentLesle** [comment]
Instead of creating a new resource why not extending the existing azurerm_express_route_circuit

**Comment by ArielRam99** [comment]
Looks like the azurerm_express_route_circuit has an arguement for express_route_port_id.

Here's the link : https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/express_route_circuit#express_route_port_id

**Comment by bardahlm** [comment]
Reading the changelogs there seems to be some progress here! New features in 2.65 and 2.66:

2.65:
* azurerm_express_route_circuit_peering - support for the bandwidth_in_gbps and express_route_port_id properties (#12289)

2.66:
* New Resouce azurerm_express_route_connection [GH-11320]
* New Resouce azurerm_express_route_circuit_connection [GH-11303]

**Comment by bardahlm** [comment]
Solved in 2.65
